fix(esp_ulp): Add support for multiple ULP program embedding without name collision (IDFGH-14152) (#14954)
espressif/esp-idf: docs/en/api-reference/system/ulp-lp-core.rst

    ulp_embed_binary(${ulp_app_name} "${ulp_sources}" "${ulp_exp_dep_srcs}")

The first argument to ``ulp_embed_binary`` specifies the ULP binary name. The name specified here is also used by other generated artifacts such as the ELF file, map file, header file, and linker export file. The second argument specifies the ULP source files. Finally, the third argument specifies the list of component source files which include the header file to be generated. This list is needed to build the dependencies correctly and ensure that the generated header file is created before any of these files are compiled. See the section below for the concept of generated header files for ULP applications.
Variables in the ULP code will be prefixed with ``ulp_`` (default value) in this generated header file.

> If you need to embed multiple ULP program, in order to avoid conflicting variable names, it's possible to modify the prefix like this:

Reviewer: Nitpick:

```suggestion
If you need to embed multiple ULP programs, you may add a custom prefix in order to avoid conflicting variable names like this:
```

    extern uint32_t ulp_measurement_count;

> Note that all symbols (variables, functions) are declared as ``uint32_t``. Array are declared as ``uint32_t [SIZE]``. For functions, take the address of the symbol and cast it to the appropriate type.

Reviewer: Nitpick:

```suggestion
Note that all symbols (variables, functions) are declared as ``uint32_t``. Arrays are declared as ``uint32_t [SIZE]``. For functions, take the address of the symbol and cast it to the appropriate type.
```

.. note::

    Variables declared in the global scope of the LP core program reside in either the ``.bss`` or ``.data`` section of the binary. These sections are initialized when the LP core binary is loaded and executed. Accessing these variables from the main program on the HP-Core before the first LP core run may result in undefined behavior.

>     The ``ulp_`` prefix is the default value. You can specify the prefix to use with ``ulp_embed_binary`` to avoid name collusions for multiple ULP program.

Reviewer: Nitpick:

```suggestion
    The ``ulp_`` prefix is the default value. You can specify the prefix to use with ``ulp_embed_binary`` to avoid name collisions for multiple ULP programs.
```
